src/gui/systray.cpp

```diff
@@ -133,123 +126,225 @@ void Systray::setToolTip(const QString &tip)
+    const auto screenRect = currentScreenRect();
+
+    qCDebug(lcSystray) << "screenRect:" << screenRect;
+    qCDebug(lcSystray) << "taskbarRect:" << taskbarRect;
+    qCDebug(lcSystray) << "taskbarScreenEdge:" << taskbarScreenEdge;
+    qCDebug(lcSystray) << "trayIconCenter:" << trayIconCenter;
+
+    switch(taskbarScreenEdge) {
+    case TaskBarPosition::Bottom:
+        return {
+            trayIconCenter.x(),
+            screenRect.bottom() - taskbarRect.height() - 4
+        };
+    case TaskBarPosition::Left:
+        return {
+            screenRect.left() + taskbarRect.width() + 4,
+            trayIconCenter.y()
+        };
+    case TaskBarPosition::Top:
+        return {
+            trayIconCenter.x(),
+            screenRect.top() + taskbarRect.height() + 4
+        };
+    case TaskBarPosition::Right:
+        return {
+            screenRect.right() - taskbarRect.width() - 4,
+            trayIconCenter.y()
+        };
     }
-#endif
+    Q_UNREACHABLE();
 }
 
-bool Systray::syncIsPaused()
+QPoint Systray::computeWindowPosition(int width, int height) const
 {
-    return _syncIsPaused;
+    const auto referencePoint = computeWindowReferencePoint();
+
+    const auto taskbarScreenEdge = taskbarOrientation();
+    const auto screenRect = currentScreenRect();
+
+    const auto topLeft = [=]() {
+        switch(taskbarScreenEdge) {
+        case TaskBarPosition::Bottom:
+            return referencePoint - QPoint(width / 2, height);
+        case TaskBarPosition::Left:
+            return referencePoint;
+        case TaskBarPosition::Top:
+            return referencePoint - QPoint(width / 2, 0);
+        case TaskBarPosition::Right:
+            return referencePoint - QPoint(width, 0);
+        }
+        Q_UNREACHABLE();
+    }();
+    const auto bottomRight = topLeft + QPoint(width, height);
+    const auto windowRect = [=]() {
+        const auto rect = QRect(topLeft, bottomRight);
+        auto offset = QPoint();
+
+        if (rect.left() < screenRect.left()) {
+            offset.setX(screenRect.left() - rect.left() + 4);
+        } else if (rect.right() > screenRect.right()) {
+            offset.setX(screenRect.right() - rect.right() - 4);
+        }
+
+        if (rect.top() < screenRect.top()) {
+            offset.setY(screenRect.top() - rect.top() + 4);
+        } else if (rect.bottom() > screenRect.bottom()) {
+            offset.setY(screenRect.bottom() - rect.bottom() - 4);
+        }
+
+        return rect.translated(offset);
+    }();
+
+    qCDebug(lcSystray) << "taskbarScreenEdge:" << taskbarScreenEdge;
+    qCDebug(lcSystray) << "screenRect:" << screenRect;
+    qCDebug(lcSystray) << "windowRect (reference)" << QRect(topLeft, bottomRight);
+    qCDebug(lcSystray) << "windowRect (adjusted )" << windowRect;
+
+    return windowRect.topLeft();
 }
 
-void Systray::pauseResumeSync()
+/// Returns a QPoint that constitutes the center coordinate of the tray icon
```

Comment:
This is a doxygen comment and should be moved to the header file to be visible, right?

Reply:
Doesn't bring much value I'll just remove it I think.